pages/locations/index.js

```diff
@@ -35,83 +35,94 @@ export default class LocationsPage extends Component {
     super(props)
     this.state = {
       locations: null,
+      locationsWithDistance: [],
       currentLocation: null,
     }
   }
 
   componentDidMount() {
     document.title = 'Link-SF'
 
-    if (navigator) {
-      navigator.geolocation.getCurrentPosition(this.setCurrentLocation)
-    } else {
-      this.setLocations()
-    }
-
     fetchLocations(20)
       .then(locations => {
-        this.setState({ locations })
+        this.setState({ locations }, this.loadRemaining)
       })
   }
 
-  setCurrentLocation = (currentLocation) => {
-    this.setState({ currentLocation }, this.setLocations)
+  setCurrentLocation = () => {
+    if (navigator) {
+      navigator.geolocation.getCurrentPosition(currentLocation => {
+        this.setState({ currentLocation }, this.addDistances)
+      })
+    }
+  }
+
+  addDistances = () => {
+    const { currentLocation, locations, locationsWithDistance } = this.state
+    const locationSlices = R.splitEvery(25, locations)
+
+    if (!currentLocation) return
+
+    this.calculateDistances(locationSlices, currentLocation)
   }
 
-  setLocations = () => {
-    const { currentLocation } = this.state
-
-    if (currentLocation && this.state.locations.length > 0) {
-      calculateAllDistances(this.state.locations, currentLocation)
-        .then(matrixResponse => {
-          const matrixResponses = matrixResponse.rows[0].elements
-          this.setState({ locations: mergeLocationsAndDistances(this.state.locations, matrixResponse) })
-        })
-    } else if (currentLocation) {
-      fetchLocations(20)
-        .then(locations => {
-          locationsCache = locations
-          return calculateAllDistances(locations, currentLocation)
-        })
-        .then(matrixResponse => {
-          const matrixResponses = matrixResponse.rows[0].elements
-          this.setState({ locations: mergeLocationsAndDistances(locationsCache, matrixResponses) })
-        })
+  calculateDistances = (locationSlices, currentLocation) => {
+    const { locationsWithDistance } = this.state
+    const a = R.splitAt(1, locationSlices)
```

Comment:
Yeah I didn't really care about that name because its only used to define the two right below it. Those ones have the meaningful names. :shrug: